content/docs/faq-state.md

```diff
@@ -6,25 +6,25 @@ layout: docs
 category: FAQ
 ---
 
-### What does `setState` do? {#what-does-setstate-do}
+### `setState` 的作用？ {#what-does-setstate-do}
 
-`setState()` schedules an update to a component's `state` object. When state changes, the component responds by re-rendering.
+`setState()` 安排對 component `state` object 的更新。當 state 改變時，component 會藉由重新 render 來回應。
 
-### What is the difference between `state` and `props`? {#what-is-the-difference-between-state-and-props}
+### `state` 和 `props` 有什麼不同？ {#what-is-the-difference-between-state-and-props}
 
-[`props`](/docs/components-and-props.html) (short for "properties") and [`state`](/docs/state-and-lifecycle.html) are both plain JavaScript objects. While both hold information that influences the output of render, they are different in one important way: `props` get passed *to* the component (similar to function parameters) whereas `state` is managed *within* the component (similar to variables declared within a function).
+[`props`](/docs/components-and-props.html)（「properties」的簡寫）和 [`state`](/docs/state-and-lifecycle.html) 都是純 JavaScript object。雖然兩者都擁有會影響 render 輸出的資訊，但在一個重要方向上有所不同：`props` 是被傳遞 *進* component（類似於 function 的參數），而 `state` 是在 component *內部* 被管理（類似於在 function 中宣告中的變數）。
 
-Here are some good resources for further reading on when to use `props` vs `state`:
+這裡有一些很棒的資源，可以進一步閱讀來了解使用 `props` 和 `state` 的時機：
 * [Props vs State](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)
 * [ReactJS: Props vs. State](https://lucybain.com/blog/2016/react-state-vs-pros/)
 
-### Why is `setState` giving me the wrong value? {#why-is-setstate-giving-me-the-wrong-value}
+### 為什麼 `setState` 會出現錯誤的值？ {#why-is-setstate-giving-me-the-wrong-value}
 
-In React, both `this.props` and `this.state` represent the *rendered* values, i.e. what's currently on the screen.
+在 React，`this.props` 和 `this.state` 都代表該 *render* 的值。也就是目前螢幕上的內容。
 
-Calls to `setState` are asynchronous - don't rely on `this.state` to reflect the new value immediately after calling `setState`. Pass an updater function instead of an object if you need to compute values based on the current state (see below for details).
+呼叫 `setState` 是非同步的 — 不要在呼叫 `setState` 後立即憑藉著 `this.state` 去反應新的值。如果需要基於目前 state 來計算值，請傳遞一個更新用的 function 而不是 object（細節請看下方）。
```

Comment:
```suggestion
呼叫 `setState` 是非同步的 — 不要在呼叫 `setState` 後立即依賴 `this.state` 去反映新的值。如果需要基於目前 state 來計算值，請傳遞一個更新用的 function 而不是 object（細節請看下方）。
```

I prefer `反映` than `反應` in here. If make confused, maybe it can keep the original term.